azure-pipelines.yml

```diff
@@ -22,6 +22,6 @@ steps:
   displayName: 'set up the Go workspace'
 
 - powershell: |
-    go test -mod=vendor -race ./...
+    go test -race ./...
```

Comment:
@marwan-at-work this one is a bit problematic because now we'll be building Athens from VCS as the upstream. Maybe we should leave in the vendor directory for now so az pipelines still works?

What do you think?

Reply:
Another option is we can run an Athens in pipelines after #1126 is done

Reply:
@arschles yes the azure fix in this PR is not intended to be merged :)

Reply:
ok, thanks not what I meant, sorry for being unclear @marwan-at-work 

I basically meant we could merge this now and then move the dependencies from GCS over to Azure after #1126 is done

Cool with that?

Reply:
@arschles hmm sorry not sure I understand. What would moving the deps from GCS to Azure fix? I imagine Azure pipelines is a containerized environment so whether the Deps are in GCS or Azure Blobs, it'll need credentials to access them. But maybe im missing something. Lemme know!

Reply:
Sorry about that @marwan-at-work. I'll try to explain more explicitly, inline:

>What would moving the deps from GCS to Azure fix?

Just the billing issue. I don't think we 100% definitely need to do this, but since we can get Azure resources for free at the moment, I figure it should be on the table. At the same time, though, I don't want that to slow _this_ PR down.

>I imagine Azure pipelines is a containerized environment so whether the Deps are in GCS or Azure Blobs, it'll need credentials to access them

Yup, you're right. But regardless of whether we do GCS or Azure blob storage, could we set permissions to read-only by the public?

Reply:
>could we set permissions to read-only by the public? Or do we then need to figure out a different way to get dependencies into storage?

@arschles I don't think so, because when we add a new dependency we need to be able to persist to the storage. We can potentially do something weird or manual to avoid it but I don't think that's necessary. My opinion is that we should give read/write permissions and ensure that they don't get exposed during CI. But I'm happy to explore alternatives of course. 

As for the cost issue, that sounds good to me. Has it been expensive? I'm happy to contribute to the cost. But also if moving to Azure Blobs removes that headache, then sounds great. Maintainers, would need some access to Azure blobs to be able to debug issues when they arise.

However, one big blocker to this is the fact that we don't have a distributed SingleFlight for Azure. But with GCP it's _both_ a Backend Storage and a Distributed SingleFlight. 

If you make an Athens Azure account, I'm happy to look more into that side of things :)

Reply:
@marwan-at-work 
> However, one big blocker to this is the fact that we don't have a distributed SingleFlight for Azure.

There's an issue for that #1140 
I'm working on it and should have something in the next few days.

Reply:
@marwan-at-work the creds thing makes sense. I was hoping there would be a way to make the bucket read only to the public, but no dice!

Cost is not a big deal at the moment. I know we haven't been building Athens with Athens yet and so storage and traffic in GCS hasn't been as it will be, but this month's bill is $0.01 so I think we should see how it goes 😄